pkg/k8s/resource_helper.go

```diff
@@ -102,6 +104,11 @@ func (rh *ResourceHelper) NewResourceFromBytes(bytes []byte) (*Resource, error)
 	// format, API group, kind, version
 	obj, _, err := scheme.Codecs.UniversalDeserializer().Decode(bytes, nil, nil)
 
+	// A missing `Kind` most probably meant an empty document
+	if runtime.IsMissingKind(err) {
```

Comment:
I wonder if we could more explicitly exclude empty documents? I think we'd still want to pick up a valid use case where someone has accidentally missed off the `Kind`

Reply:
@milesbxf Decode function in `scheme.Codecs.UniversalDeserializer().Decode(bytes, nil, nil)` decodes the yaml and immediately looks for a `Kind` before doing anything else, there doesn't seem to be a more specific error than this one unfortunately.  

https://github.com/kubernetes/apimachinery/blob/master/pkg/apis/meta/v1/unstructured/helpers.go#L326-L344

Reply:
Urgh, that's annoying. Perhaps we could directly decode the YAML ourselves, exclude the empty documents, then pass to the decoder as bytes?

Reply:
@milesbxf We could do a `gvk.Empty()` as well, will that work? Found this. 

```
// Empty returns true if group, version, and kind are empty
func (gvk GroupVersionKind) Empty() bool {
	return len(gvk.Group) == 0 && len(gvk.Version) == 0 && len(gvk.Kind) == 0
}
```

Reply:
Hmm - is `gvk` a pointer? If not, `gvk` will be empty whenever there is any kind of error